### DISCUSSION TITLE
Access parent model of preloaded relationship

### DISCUSSION THREAD

**Discussion by WeyIin** [adonisjs/core#3537] [Help]
Hi folks

So here's the question if anyone was able to get it solved or around it:
Two models, Person and Stats.

Relationship is 1 to 1.
In Person
```
  @hasOne(() => Stat)
  public stats: HasOne<typeof Stat>
```
In Stat
```
  @belongsTo(() => Person)
  public person: BelongsTo<typeof Person>
```

In stat model I have a computed property:
```
@computed()
public get modifiedTitle() {
  if(this.person.title == 'normaluser') {
    // do something 
}
}
```

For the above to work, I have to preload stat and person again which seem to be a bit dumb - otherwise `this.person` is null (since it is not preloaded)

e.g. 

```
public async showProfile({ auth }: HttpContextContract) {
      const person = auth.user
     person.load('stats')

   /// person.stats.modifiedTitle is null since `this.person` is null in the stat model.
//this however works..

        const person = auth.user
     person.load('stats', (person) => {
       person.preload('person')
})

}
```

If I output what person object gives me, it is person object having stats loaded and within stats, I have same person details loaded again. 

My question is, how can I access `this.person` inside stat model without going back and forth preloading relationship that is already defined and pretty much loaded? since stat is being loaded by person anyway. Why do I need to go back? Anyway to get person object without preloading it back?

**Comment by thetutlage** [comment]
What you are expecting is not possible and neither it can be added to Lucid, since it makes things a lot less predictable. 

However, you can solve it quite easily within your controller.

```ts
const person = auth.user!

await person.load('stats')
person.stats?.person = person
```

If you want to keep it DRY, then you can simply add a method to your model as

```ts
class Person extends BaseModel {
  public async loadStats() {
    const self = this as Person
    await self.load('stats')
    self.stats?.person = self
  }
}

await person.loadStats()
```

**Comment by WeyIin** [comment]
Thanks @thetutlage - I already have something in place in regards to auto preloading, I just thought if there's something to get the data rather than preloading it all over again.. because let's say we use on query preload, this can potentially go in a inf loop.

E.g. 
Auto preload stats from person and person from stats and back again..

What is the impact on the database? Is that cached if values are unchanged or it is hitting the database every time? 

Sorry  I haven't done any investigation in regards to this.
